src/supply.ts

```diff
@@ -27,57 +32,66 @@ export const getBalance = async (
           .map(([key, value]) => `${key}=${value}`)
           .join("&")}`;
       }
-      const holders = await fetch(url);
-      const json: any = await holders.json();
-      const _addresses = json.items.map((holder: any) => holder.address.hash);
-      addresses.push(..._addresses);
-      if (!json.next_page_params) {
-        break;
+      
+      try {
+        const holdersResponse = await fetch(url);
+        if (!holdersResponse.ok) {
+          throw new Error(`Failed to fetch data. Status: ${holdersResponse.status}`);
+        }
+        
+        const json:any = await holdersResponse.json();
+        const _addresses = json.items.map((holder: any) => holder.address.hash);
+        addresses.push(..._addresses);
+        
+        if (!json.next_page_params) {
+          break; // Exit the loop if there are no more pages
+        }
+        
+        nextPageParams = json.next_page_params;
+        console.log("nextPageParams: ", nextPageParams);
+        console.log("fetching next page...");
+      } catch (error) {
+        console.error("Error fetching token holders:", error);
+        break; // Exit the loop on error
```

Comment:
Should not simply break here, since it will give us incomplete results. We need to only break when we know for sure we have reached the last page. Instead of break we should return an error response from the API, maybe a 400 error.